messaging/src/test/java/org/axonframework/messaging/unitofwork/StubProcessingContext.java

```diff
@@ -167,15 +201,75 @@ public static ProcessingContext forMessage(Message<?> message) {
     }
 
     /**
-     * Creates a new stub {@link ProcessingContext} for the given {@link LegacyUnitOfWork}.
-     * You can use this to create a context compatible with most of the framework.
-     * Do note that this context does not commit or advance phases on its own,
+     * Creates a new stub {@link ProcessingContext} with the given {@code component}. You can use this to
+     * create a context compatible with most of the framework. Do note that this context does not commit or advance
+     * phases on its own, but you can use {@link #moveToPhase(Phase)} to advance the context to a specific phase.
+     *
+     * @param type     The type of the component to register.
+     * @param instance The instance of the component to register.
+     * @param <C>      The type of the component to register.
+     * @return A new {@link ProcessingContext} instance containing the given {@code component} as a resource.
+     */
+    public static <C> StubProcessingContext withComponent(@Nonnull Class<C> type, @Nonnull C instance) {
```

Comment:
Ow sweet, good addition for sure. One point though: won't consecutive invocations of either of these `withComponent` methods override previously registered components? Wouldn't you want to be able to chain that?

Reply:
It's a static method, so there is nothing to chain with. Or am I missing something?